### DISCUSSION TITLE
Dissolve effect in o3de?!!

### DISCUSSION THREAD

**Discussion by watchmax3** [o3de/o3de#8099] [General]
Do that with landscape canvas?!! Material editor?! Where can work with shaders? First I looking for something like build modifier of blender but dissolve was easier but I don't know where create dissolve in o3de? ( dissolve for meshes)( dissolve FX parallax occlusion mapping... In lumberyard are rip now in o3de)

**Comment by santorac** [comment]
What kind of dissolve effect are you looking for? I'm not sure what you mean.

As far as working with shaders, at this time they are all manually authored, though we (AWS) have plans to work on a node graph material authoring tool.

You mentioned parallax occlusion mapping; this is supported by the StandardPBR and EnhancedPBR material types. There is also StandardMultilayerPBR that supports multiple layers blended together with parallax.

**Comment by watchmax3** [comment]
I got it. So they  will create a tool for that. I looking for usual dissolve wich exist in almost video games (after death of characters) thanks for answer and help.

**Comment by santorac** [comment]
Oh, so like a full screen dissolve to black, or dissolve to a UI screen, or something like that?

**Comment by santorac** [comment]
Check out this tutorial video, you might find this helpful: https://www.youtube.com/watch?v=cqQNHGkyo94

**Comment by HogJonny-AMZN** [comment]
Here is a guide I wrote that is a step by step with the smallest number of changes to add a new pass via files within my game project:
[ATOM-126432892-290322-1426-202.pdf](https://github.com/o3de/o3de/files/8372387/ATOM-126432892-290322-1426-202.pdf)
[Passes.zip](https://github.com/o3de/o3de/files/8372390/Passes.zip)
[Shaders.zip](https://github.com/o3de/o3de/files/8372391/Shaders.zip)

This would allow you to do a fullscreen pass where you could do some kind of dissolve Post FX.

But @santorac I think what is being asked for here is a per-object dissolve. That would be done via a shader, and it's often secondary opacity. 
1. normal opacity, with a alpha mask, determines the objects clipping/blending in texture space
2. dissolve is an opacity in another space, like the ability to dissolve from top to bottom

Ways I have done this:
1. you need a gradient to use for the dissolve, that could be some combination of another alpha texture, vertex color channel, or some kind of gradient derived from object or world space (like using the bounding box)
2.  drive clipping along this gradient via a threshold
3. add a dithered band within this gradient
4. supplement pixel-frag shading by darkening based on a band within the gradient, or make the leading edge burn with a glow

Extending these ideas, a phaser hit could be the point of origin, and the gradient could be a radius. 

Right now, this type of approach would either be a customized shader materialtype, or wait for Material and Shader Canvas

**Reply by watchmax3** [reply]
Thanks for answer and help. I will try your solutions

**Comment by HogJonny-AMZN** [comment]
@watchmax3 Landscape Canvas does have visual nodes and a composition node graph but that system currently is in world space, it is not a traditional texture graph. It's for terrain and vegetation landscaping systems primarily.

**Reply by watchmax3** [reply]
So. It s just for terrain and vegetation.
